http/rest-client-reactive-vanilla/src/test/java/io/quarkus/ts/http/restclient/reactive/VanillaReactiveRestClientIT.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class VanillaReactiveRestClientIT {
+    private static final String CONTAINER_FILTER_ERROR_TEXT = "CDI: programmatic lookup problem detected";
+
+    @QuarkusApplication
+    static RestService app = new RestService();
+
+    @Test
+    @Tag("https://github.com/quarkusio/quarkus/issues/31024")
+    public void restClientContainerFilterTest() {
+        List<String> logs = app.getLogs();
```

Comment:
IMO you should add Vert.x HTTP extension and call the route which uses REST client to call another route to receive response. Then you have end to end test.

Reply:
Another argument I can think of is not hanging on log message when not necessary. Problem is that the message can be slightly altered and we would never know (which means test would not fail with a regression).

Reply:
As stated below, we agreed that there is no better way, than to look into logs.